### PR TITLE
feat: retry remove finalizer

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

**Review by metacosm — CHANGES_REQUESTED** [review]
LGTM apart from some confusion on some of the comments / doc

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java [hunk 1]
```diff
@@ -35,6 +37,8 @@
  */
 class ReconciliationDispatcher<R extends HasMetadata> {
 
+  public static final int MAX_FINALIZER_REMOVAL_RETRY = 10;
```
[the remark this answers is not in the document]
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java:40]: Would do it a separate PR TBH. Ideally this retry passes after 1-2 attemt, it's unlikely that there will be more than 3-4 finalizers.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventMarker.java [hunk 2]
```diff
@@ -19,7 +20,7 @@ class EventMarker {
 
   public enum EventingState {
     /** Event but NOT Delete event present */
```
[the remark this answers is not in the document]
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventMarker.java]: fixed

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java:40]: Yes, this should be done in a different PR for sure.

**Review by metacosm — APPROVED** [review]

**Pull request merged**